File: app/src/main/java/com/riders/thelab/data/local/model/weather/CityModel.java
package com.riders.thelab.data.local.model.weather;

import android.database.Cursor;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import com.riders.thelab.data.remote.dto.weather.City;

import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Entity(tableName = "city")
@Setter
@Getter
@ToString
public class CityModel {

    @PrimaryKey
    @NonNull
    @ColumnInfo(name = "_id")
    public int id;
    @ColumnInfo(name = "name")
    public String name;
    @ColumnInfo(name = "state")
    public String state;
    @ColumnInfo(name = "country")
    public String country;
    @ColumnInfo(name = "longitude")
    public double longitude;
    @ColumnInfo(name = "latitude")
    public double latitude;

    public CityModel() {
    }

    @Ignore
    public CityModel(City dtoCity) {
        // TODO create mapper
        setId(dtoCity.getId());
        setName(dtoCity.getName());
        setState(dtoCity.getState());
        setCountry(dtoCity.getCountry());
        setLatitude(dtoCity.getCoordinates().getLatitude());
        setLongitude(dtoCity.getCoordinates().getLongitude());
    }

    @Ignore
    public CityModel(Cursor cursor) {
        setName(cursor.getString(cursor.getColumnIndexOrThrow("name")));
        setState(cursor.getString(cursor.getColumnIndexOrThrow("state")));
        setCountry(cursor.getString(cursor.getColumnIndexOrThrow("country")));
        setLatitude(cursor.getDouble(cursor.getColumnIndexOrThrow("latitude")));
        setLongitude(cursor.getDouble(cursor.getColumnIndexOrThrow("longitude")));
    }
}

File: app/src/main/java/com/riders/thelab/data/local/model/weather/CityModel.kt
package com.riders.thelab.data.local.model.weather

import android.database.Cursor
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.riders.thelab.data.remote.dto.weather.City

@Entity(tableName = "city")
data class CityModel constructor(
    @PrimaryKey
    @ColumnInfo(name = "_id")
    var id: Int = 0,

    @ColumnInfo(name = "name")
    var name: String,

    @ColumnInfo(name = "state")
    var state: String,

    @ColumnInfo(name = "country")
    var country: String,

    @ColumnInfo(name = "longitude")
    var longitude: Double = 0.0,

    @ColumnInfo(name = "latitude")
    var latitude: Double = 0.0
) {
    constructor() : this(-1, "", "", "", 0.0, 0.0)

    @Ignore
    constructor(dtoCity: City) : this() {
        this.id = dtoCity.id
        this.name = dtoCity.name
        this.state = dtoCity.state
        this.country = dtoCity.country
        this.longitude = dtoCity.coordinates.longitude
        this.latitude = dtoCity.coordinates.latitude
    }

    @Ignore
    constructor(cursor: Cursor) : this() {
        this.name = cursor.getString(cursor.getColumnIndexOrThrow("name"))
        this.state = (cursor.getString(cursor.getColumnIndexOrThrow("state")))
        this.country = (cursor.getString(cursor.getColumnIndexOrThrow("country")))
        this.latitude = (cursor.getDouble(cursor.getColumnIndexOrThrow("latitude")))
        this.longitude = (cursor.getDouble(cursor.getColumnIndexOrThrow("longitude")))
    }

    override fun toString(): String {
        return "CityModel(id=$id, name=$name, state=$state, country=$country, longitude=$longitude, latitude=$latitude)"
    }
}
